practice-two/src/components/ShoppingCart.jsx

```diff
@@ -53,29 +56,37 @@ const ShoppingCart = ({ products = [] }) => {
         >
           <div className="w-full h-full">
             <div className="inline-block">
-              <Typography level={2} size="xl" variant="soild">
+              <Typography level={2} size="xl" variant="plain">
                 My Cart
               </Typography>
             </div>
-            {/* TODO: Update the display of the cart when there are no products */}
-            {/* <div className="mt-20 flex w-full flex-col items-center justify-center overflow-hidden">
-              <img src={ICON.cart} alt="cart-icon" className="w-24" />
-              <Typography className="mt-6 text-center text-2xl font-bold">Your cart is empty.</Typography>
-            </div> */}
-            <div className="flex h-full w-full flex-col justify-between overflow-hidden p-1 pb-10">
-              <ul className="flex-grow overflow-auto py-4">
-                {products.map((product) => (
-                  <CartItem key={product.id} product={product} />
-                ))}
-              </ul>
-              <TotalSection />
-              <Button
-                variant="outline"
-                customVariant="inline-block bg-secondary hover:bg-blue-800 focus:bg-blue-800 py-2 px-4 rounded-3xl text-lg text-white font-semibold flex justify-center"
-              >
-                Proceed to Checkout
-              </Button>
-            </div>
+            {cartProductIDs.length === 0 ? (
+              <div className="mt-20 flex w-full flex-col items-center justify-center overflow-hidden">
+                <img src={ICON.cart} alt="cart-icon" className="w-24" />
+                <Typography variant="plain" className="mt-6 text-center text-2xl font-bold">
+                  Your cart is empty.
+                </Typography>
+              </div>
+            ) : (
+              <div className="flex h-full w-full flex-col justify-between overflow-hidden p-1 pb-10">
+                <ul className="flex-grow overflow-auto py-4">
+                  {cartProductIDs.map((productId) => {
```

Comment:
Why do find in map and return null ??? Could you please refactor it

Reply:
Yes, Removed

Reply:
check it plz